DataFormats/EcalDigi/src/EcalTimeDigi.cc

```diff
@@ -10,11 +10,9 @@ EcalTimeDigi::EcalTimeDigi(const DetId& id)
     : id_(id), size_(0), sampleOfInterest_(-1), waveform_(WAVEFORMSAMPLES), data_(MAXSAMPLES) {}
 
 void EcalTimeDigi::setSize(unsigned int size) {
+  size_ = size;
   if (size > MAXSAMPLES)
-    size_ = MAXSAMPLES;
-  else
-    size_ = size;
-  data_.resize(size_);
+    data_.resize(size_);
```

Comment:
I just want to note that according to https://github.com/cms-sw/cmssw/issues/40764#issuecomment-1507230611 the silent cap of `MAXSAMPLES` does not seem to be the cause for https://github.com/cms-sw/cmssw/issues/40764